packages/react/src/primitives/Checkbox/Checkbox.tsx

```diff
@@ -26,26 +27,51 @@ const CheckboxPrimitive: Primitive<CheckboxProps, 'input'> = (
     label,
     labelHidden,
     labelPosition,
-    onChange: onChangeProp,
+    onBlur: _onBlur,
+    onFocus: _onFocus,
+    onChange: _onChange,
     testId,
     inputStyles,
     ..._rest
   },
   ref
 ) => {
   const { styleProps, rest } = splitPrimitiveProps(_rest);
-  // controlled way should always override uncontrolled way
-  const initialChecked = checked !== undefined ? checked : defaultChecked;
 
-  const { dataChecked, dataFocus, onBlur, onChange, onFocus, setDataChecked } =
-    useCheckbox(initialChecked, onChangeProp);
+  const [focused, setFocused] = React.useState(false);
 
-  React.useEffect(() => {
-    const isControlled = checked !== undefined;
-    if (isControlled && checked !== dataChecked) {
-      setDataChecked(checked);
```

Comment:
Removed this effect, it is unnecessary to update the local checked state if the `Checkbox` is controlled